### PR TITLE
[Workers] KV wrangler guide

### PR DESCRIPTION
**Pull request by deadlypants1973**
Instructions on how to use KV with Wrangler and through the dashboard.

### CODE REVIEW

File: content/workers/wrangler/workers-kv.md [hunk 1]
```diff
@@ -4,17 +4,30 @@ title: Workers KV
...
+## Get started
+
+You can interact with KV via:
+
+* [Wrangler(command line)]().
+* The [Cloudflare dashboard]().
+
+### Get started with Wrangler
+
+Wrangler allows you to create, list, and delete KV namespaces. It also allows you to `PUT`, `LIST`, `GET`, and `DELETE` entries within your KV namespace.
```
Comment (koeninger) [inline content/workers/wrangler/workers-kv.md]: capitalization of  PUT etc doesn't make sense here - neither wrangler instructions nor the workers js api capitalize those commands.

Comment (deadlypants1973) [reply content/workers/wrangler/workers-kv.md]: thank you! will edit this

File: content/workers/wrangler/workers-kv.md [hunk 2]
```diff
@@ -12,59 +12,91 @@ To learn more about how KV works, refer to [How KV works](/workers/learning/how-
...
+    { binding = "<YOUR_BINDING>", id = "<YOUR_ID>" }
+]
 ```
 
-To write a value to your KV namespace using Wrangler, run the `wrangler kv:key put` subcommand:
+Note that binding names do not need to correspond to the namespace you created. It is an entirely new value that you assign.
+
+{{<Aside type="note" header="Bindings">}}
+
+A binding is a how your Worker interacts with external resources such as [KV Namespaces](/workers/runtime-apis/kv/), [Durable Objects](/workers/runtime-apis/durable-objects/), or [R2 Buckets](#api). A binding is a runtime variable that the Workers runtime provides to your code. You can declare a variable name in your `wrangler.toml` file that will be bound to these resources at runtime, and interact with them through this variable. Every binding's variable name and behavior is determined by you when deploying the Worker. Refer to the [Environment Variables](https://github.com/workers/platform/environment-variables) documentation for more information.
```
Comment (koeninger) [inline content/workers/wrangler/workers-kv.md]: environment variables link shouldn't be to github.com, R2 buckets link looks like it's not set

File: content/workers/wrangler/workers-kv.md [hunk 3]
```diff
@@ -4,54 +4,101 @@ title: Workers KV
...
 
-KV operations are scoped to your account, so in order to use any of these commands, you must:
+To learn more about how KV works, refer to [How KV works](/workers/learning/how-kv-works).
 
-1. Configure an `account_id` in your project's `wrangler.toml` file or set the `CLOUDFLARE_ACCOUNT_ID` environment variable.
-2. Run all `wrangler kv:<COMMAND>` operations in your terminal from the project's root directory.
+To review the API spec, refer to the [KV in Runtime API](/workers/runtime-apis/kv).
 
-### Get started
+You can interact with KV via Wrangler and the Cloudflare dashboard. This guide
```
Comment (koeninger) [inline content/workers/wrangler/workers-kv.md]: "This guide" looks like a sentence fragment?

File: content/workers/wrangler/workers-kv.md [hunk 4]
```diff
@@ -4,54 +4,101 @@ title: Workers KV
...
+
+Wrangler allows you to create, list, and delete KV namespaces. It also allows you to put, list, get, and delete entries within your KV namespace.
+
+KV operations are scoped to your account. To use Workers KV with your Worker, the first thing you must do is create a KV namespace. This is done with the `kv:namespace` subcommand. 
+
+To create a KV namespace with Wrangler:
+
+1. Run `wrangler kv:namespace create <YOUR_NAMESPACE>` in your terminal.
 
 The `kv:namespace` subcommand takes as a new binding name as its argument. A Workers KV namespace will be created using a concatenation of your Worker’s name (from your `wrangler.toml` file) and the binding name you provide:
```
Comment (koeninger) [inline content/workers/wrangler/workers-kv.md]: "takes as a new binding name" -> "takes a new binding name"

File: content/workers/wrangler/workers-kv.md [hunk 5]
```diff
@@ -65,32 +112,19 @@ kv_namespaces = [
 ]
 ```
 
-To insert a value into a specific KV namespace, use:
+Using the same binding name for two different namespaces keeps your Worker code simple. You can set a `.get()` command in your Worker code for one binding but when the Worker is running (`let value = await <YOUR_BINDING>.get("KEY");`), data distribution flows according to the environments set up in your `wrangler.toml` file.
```
Comment (koeninger) [inline content/workers/wrangler/workers-kv.md]: "data distribution flows according to the environments" isn't very clear to me.

I think it'd be clearer to explicitly say something along the lines of 
In the staging environment, `MY_KV.get("KEY")` will read from the namespace id `e29b263ab50e42ce9b637fa8370175e8`. In the production environment, `MY_KV.get("KEY")` will read from the namespace id `a825455ce00f4f7282403da85269f8ea`

**Comment by koeninger** [comment]
Mostly small textual comments.  Bigger picture, I think the most confusing things here are likely to be wrangler1 environments vs service environments, and global environment variables vs module workers taking an env parameter.  The examples here all seem to be for the older versions of each, is there a concrete plan for updating them?

File: content/workers/wrangler/workers-kv.md (hunk 3 above)
Comment (deadlypants1973) [inline content/workers/wrangler/workers-kv.md]:
```suggestion
You can interact with KV via Wrangler and the Cloudflare dashboard. This page will guide you through creating your KV namespace, writing a new value to your namespace and interacting with [environments](/workers/platform/environments/).
```

File: content/workers/wrangler/workers-kv.md (hunk 4 above)
Comment (deadlypants1973) [inline content/workers/wrangler/workers-kv.md]:
```suggestion
The `kv:namespace` subcommand takes a new binding name as its argument. A Workers KV namespace will be created using a concatenation of your Worker’s name (from your `wrangler.toml` file) and the binding name you provide:
```

File: content/workers/wrangler/workers-kv.md [hunk 6]
```diff
@@ -4,54 +4,101 @@ title: Workers KV
...
+    { binding = "<YOUR_BINDING>", id = "<YOUR_ID>" }
+]
 ```
 
-To write a value to your KV namespace using Wrangler, run the `wrangler kv:key put` subcommand:
+Note that binding names do not need to correspond to the namespace you created. It is an entirely new value that you assign.
+
+{{<Aside type="note" header="Bindings">}}
+
+A binding is a how your Worker interacts with external resources such as [KV Namespaces](/workers/runtime-apis/kv/), [Durable Objects](/workers/runtime-apis/durable-objects/), or [R2 Buckets](#api). A binding is a runtime variable that the Workers runtime provides to your code. You can declare a variable name in your `wrangler.toml` file that will be bound to these resources at runtime, and interact with them through this variable. Every binding's variable name and behavior is determined by you when deploying the Worker. Refer to the [Environment Variables](https://github.com/workers/platform/environment-variables) documentation for more information.
```
Comment (deadlypants1973) [inline content/workers/wrangler/workers-kv.md]:
```suggestion
A binding is a how your Worker interacts with external resources such as [KV Namespaces](/workers/runtime-apis/kv/), [Durable Objects](/workers/runtime-apis/durable-objects/), or [R2 Buckets](/r2/runtime-apis/). A binding is a runtime variable that the Workers runtime provides to your code. You can declare a variable name in your `wrangler.toml` file that will be bound to these resources at runtime, and interact with them through this variable. Every binding's variable name and behavior is determined by you when deploying the Worker. Refer to the [Environment Variables](/workers/platform/environment-variables) documentation for more information.
```

File: content/workers/wrangler/workers-kv.md (hunk 5 above)
Comment (deadlypants1973) [inline content/workers/wrangler/workers-kv.md]:
```suggestion
Using the same binding name for two different namespaces keeps your Worker code simple. In the `staging` environment, `MY_KV.get("KEY")` will read from the namespace ID `e29b263ab50e42ce9b637fa8370175e8`. In the `production` environment, `MY_KV.get("KEY")` will read from the namespace ID `a825455ce00f4f7282403da85269f8ea`.
```

**Review by mmalden — APPROVED** [review]

**Comment by deadlypants1973** [comment]
> Mostly small textual comments. Bigger picture, I think the most confusing things here are likely to be wrangler1 environments vs service environments, and global environment variables vs module workers taking an env parameter. The examples here all seem to be for the older versions of each, is there a concrete plan for updating them?

I am working on this with Kabir and Sunil. Waiting on the service environments release before making new changes.

**Pull request merged**